b2b_ecommerce/models.py

```diff
@@ -0,0 +1,102 @@
+"""Models for business to business ecommerce"""
+import uuid
+
+from django.conf import settings
+from django.contrib.postgres.fields import JSONField
+from django.db import models
+
+from b2b_ecommerce.constants import REFERENCE_NUMBER_PREFIX
+from ecommerce.models import (
+    CouponPaymentVersion,
+    ProductVersion,
+    OrderAbstract,
+    OrderManager,
+)
+from mitxpro.models import AuditModel, AuditableModel, TimestampedModel
+from mitxpro.utils import serialize_model_object
+
+
+class B2BOrder(OrderAbstract, AuditableModel):
+    """
+    An order containing information for the purchase of enrollment codes by businesses or other bulk purchasers.
+    Orders which are fulfilled represent successful completion of a purchase and are the source of truth
+    for this information.
+    """
+
+    num_seats = models.PositiveIntegerField()
+    email = models.EmailField()
+    product_version = models.ForeignKey(ProductVersion, on_delete=models.PROTECT)
+    per_item_price = models.DecimalField(decimal_places=2, max_digits=20)
+    total_price = models.DecimalField(decimal_places=2, max_digits=20)
+    unique_id = models.UUIDField(default=uuid.uuid4)
+    coupon_payment_version = models.ForeignKey(
+        CouponPaymentVersion, null=True, on_delete=models.PROTECT
+    )
+
+    objects = OrderManager(REFERENCE_NUMBER_PREFIX)
+
+    @property
+    def reference_id(self):
```

Comment:
This is identical in both models. Could this just be moved to `OrderAbstract`? I suggested implementing the prefix as a model class property in another comment. If you do that I think it would work